packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts

```diff
@@ -513,6 +513,487 @@ export class CloudFormationDeleteStackAction extends CloudFormationDeployAction
+ * the stack set’s AWS CloudFormation template. When you create the stack set, you specify the template to use, as well as any parameters and
+ * capabilities that the template requires.
+ *
+ * For more information about concepts for AWS CloudFormation StackSets, see StackSets concepts in the AWS CloudFormation User Guide.
+ */
+export class CloudFormationStackSetAction extends Action {
+  private static getArtifacts(props: CloudFormationStackSetActionProps) {
+    const artifacts: codepipeline.Artifact[] = [props.templatePath.artifact];
+
+    if (props.parametersPath !== undefined) {
+      artifacts.push(props.parametersPath.artifact);
+    }
+    if (props.deploymentTargetsPath !== undefined) {
+      artifacts.push(props.deploymentTargetsPath.artifact);
+    }
+
+    // de-dupe
+    return artifacts.filter((artifact, index, self) => self.indexOf(artifact) === index);
+  }
+
+  private readonly props: CloudFormationStackSetActionProps;
+
+  constructor(props: CloudFormationStackSetActionProps) {
+    super({
+      ...props,
+      provider: 'CloudFormationStackSet',
+      category: codepipeline.ActionCategory.DEPLOY,
+      artifactBounds: {
+        minInputs: 1,
+        maxInputs: 3,
+        minOutputs: 0,
+        maxOutputs: 0,
+      },
+      inputs: CloudFormationStackSetAction.getArtifacts(props),
+    });
+
+    this.props = props;
+  }
+
+  private get parameters(): string | undefined {
+    const parameters = this.props.parameters;
+    const parametersPath = this.props.parametersPath;
+
+    if (parameters !== undefined && parametersPath !== undefined) {
+      throw new Error('Cannot use both parameters and parametersPath.');
+    }
+    if (parametersPath !== undefined) {
+      return parametersPath.location;
+    }
+
+    if (parameters === undefined) {
+      return undefined;
+    }
+
+    return parameters.map((parameter) => {
+      let ret = 'ParameterKey=' + parameter.parameterKey;
+      if (parameter.parameterValue !== undefined) {
+        ret += ',ParameterValue=' + parameter.parameterValue;
+      }
+      if (parameter.usePreviousValue !== undefined) {
+        ret += ',UsePreviousValue=' + parameter.usePreviousValue;
+      }
+
+      return ret;
+    }).join(' ');
+  }
+
+  protected bound(scope: Construct, _stage: codepipeline.IStage, options: codepipeline.ActionBindOptions): codepipeline.ActionConfig {
+    const singletonPolicy = SingletonPolicy.forRole(options.role);
+    singletonPolicy.grantCreateUpdateStackSet(this.props);
+
+    if (this.props.administrationRole !== undefined) {
+      singletonPolicy.grantPassRole(this.props.administrationRole);
+    } else if (this.props.permissionModel !== StackSetPermissionModel.SERVICE_MANAGED) {
+      singletonPolicy.grantPassRole(cdk.Stack.of(scope).formatArn({
+        region: '',
+        service: 'iam',
+        resource: 'role',
+        resourceName: 'AWSCloudFormationStackSetAdministrationRole',
+      }));
```

Comment:
Instead of this, just use `Role.fromRoleName()`. This will also save you changing the signature of `grantPassRole()` below.

Reply:
I was originally going to do that (well, fromRoleArn) - no issue creating a construct within the bound method?

Reply:
Nope.